templates/src/ruby_objs.c.erb

```diff
@@ -10,107 +10,17 @@
-  "rbs_type_name",
-  "rbs_class_instance",
-  "rbs_class_singleton",
-  "rbs_alias",
-  "rbs_interface",
-  "rbs_union",
-  "rbs_intersection",
-  "rbs_tuple",
-  "rbs_optional",
-  "rbs_block",
-  "rbs_function_param",
-  "rbs_untyped_function",
-  "rbs_proc",
-  "rbs_void",
-  "rbs_literal",
-  "rbs_record",
-  "rbs_variable",
-  "rbs_method_type",
-  "rbs_ast_comment",
-  "rbs_ast_annotation",
-  "rbs_ast_type_param",
-  "rbs_ast_decl_constant",
-  "rbs_ast_decl_global",
-  "rbs_ast_decl_type_alias",
-  "rbs_ast_decl_interface",
-  "rbs_ast_decl_module_self",
-  "rbs_ast_decl_module",
-  "rbs_ast_decl_class_alias",
-  "rbs_ast_decl_module_alias",
-  "rbs_ast_members_method_definition_overload",
-  "rbs_ast_members_method_definition",
-  "rbs_ast_members_variable",
-  "rbs_ast_members_mixin",
-  "rbs_ast_members_attribute",
-  "rbs_ast_members_visibility",
-  "rbs_ast_members_alias",
-  "rbs_ast_decl_class_super",
-  "rbs_ast_decl_class",
-  "rbs_ast_directives_use",
-  "rbs_ast_directives_use_single_clause",
-  "rbs_ast_directives_use_wildcard_clause",
-]
-
-# This list contains the functions that use "args" for their local variable name, instead of "kwargs".
-# We'll use this in our auto-generated code to the existing naming, to minimize the diff.
-# It's temporary and will be removed in a follow-up PR.
-uses_args_hash = [
-  "rbs_base_type",
-  "rbs_namespace",
-  "rbs_type_name",
-  "rbs_class_instance",
-  "rbs_class_singleton",
-  "rbs_interface",
-  "rbs_union",
-  "rbs_intersection",
-  "rbs_optional",
-  "rbs_untyped_function",
-  "rbs_ast_type_param",
-  "rbs_ast_decl_constant",
-  "rbs_ast_decl_global",
-  "rbs_ast_decl_type_alias",
-  "rbs_ast_decl_interface",
-  # "rbs_ast_decl_module_self",
-  "rbs_ast_decl_module",
-  "rbs_ast_decl_class_alias",
-  "rbs_ast_decl_module_alias",
-  "rbs_ast_members_method_definition_overload",
-  "rbs_ast_members_method_definition",
-  "rbs_ast_members_alias",
-  "rbs_tuple",
-  "rbs_block",
-  "rbs_function_param",
-  "rbs_proc",
-  "rbs_literal",
-  "rbs_record",
-  "rbs_variable",
-  "rbs_method_type",
-  "rbs_ast_comment",
-  "rbs_ast_annotation"
-]-%>
-
-<%- nodes.sort_by { |node| ordered_function_names.index(node.c_function_name) || 999 }.each do |node| -%>
-VALUE <%= node.c_function_name %>(<%= node.fields.map { |field| "#{field.c_type} #{field.c_name}" }.join(", ") %>) {
-  <%- hash_var = uses_args_hash.include?(node.c_function_name) ? "args" : "kwargs" -%>
-  <%- hash_var = "kw_args" if node.c_function_name == "rbs_ast_decl_module_self" -%>
-  VALUE <%= hash_var %> = rb_hash_new();
+<%- nodes.each do |node| -%>
+VALUE <%= node.c_function_name %>(<%= node.fields.map { |field| "#{field.c_type} #{field.name}" }.join(", ") %>) {
+  VALUE _init_kwargs = rb_hash_new();
   <%- node.fields.each do |field| -%>
-  rb_hash_aset(<%= hash_var %>, ID2SYM(rb_intern("<%= field.name %>")), <%= field.c_name %>);
+  rb_hash_aset(_init_kwargs, ID2SYM(rb_intern("<%= field.name %>")), <%= field.name %>);
```

Comment:
We picked `_init_kwargs` instead of a name like `kwargs`, `args`, etc. to make it less likely that it collides with the name of a parameter to this function.